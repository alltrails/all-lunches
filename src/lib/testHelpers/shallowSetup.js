import React from 'react';
import { shallow } from 'enzyme';

export default (Component, defaultProps) =>
  (otherProps = {}) => {
    const props = { ...defaultProps, ...otherProps };
    const wrapper = shallow(<Component {...props} />);

    return { props, wrapper };
  };
