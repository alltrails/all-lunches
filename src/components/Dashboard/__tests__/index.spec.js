/* eslint-env jest */

import shallowSetup from 'lib/testHelpers/shallowSetup';

import { Dashboard } from '../index';

describe('Dashboard', () => {
  const defaultProps = { isTouchScreen: false };
  const setup = shallowSetup(Dashboard, defaultProps);

  it('renders the expected view', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view when a touch screen', () => {
    const { wrapper } = setup({ isTouchScreen: true });
    expect(wrapper).toMatchSnapshot();
  });
});
