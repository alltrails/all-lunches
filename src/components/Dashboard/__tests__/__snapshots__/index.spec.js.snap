// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders the expected view 1`] = `
<style__LayoutWrapper>
  <Default />
</style__LayoutWrapper>
`;

exports[`Dashboard renders the expected view when a touch screen 1`] = `
<style__LayoutWrapper>
  <Mobile />
</style__LayoutWrapper>
`;
